runtime/src/main/java/org/corfudb/runtime/collections/CorfuQueue.java

```diff
@@ -123,11 +114,11 @@ public int compareTo(CorfuRecordId o) {
          * @return
          */
         public boolean equals(CorfuRecordId o) {
-            return id.getLeastSignificantBits() == o.id.getLeastSignificantBits();
+            return (epoch == o.getEpoch()) && (sequence == o.getSequence()) && entryId == o.getEntryId();
```

Comment:
When you implement `equals` method you have to implement a hashCode method, if not then functionality will be broken.

Reply:
done. thanks!